src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+goog.module('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * A class that processes events pushed to the data layer
+ * by constructing and sending Google Analytics events via
+ * Measurement Protocol.
+ * For its Google Analytic events, it generates a unique client ID
+ * and stores it in the long term storage model provided.
+ * @implements {EventProcessor}
+ */
+class GoogleAnalyticsEventProcessor {
+  /**
+   * @param {{
+   *     'api_secret': (string|undefined),
+   *     'measurement_id': (string|undefined),
+   *     'measurement_url': string,
+   *     'client_id_expires': number,
+   *     'automatic_params': !Object<string, boolean>,
+   * }=} optionsObject
```

Comment:
It seems to me like the first 2 parameters are required, so they can't be undefined. And the last 3 parameters are optional, so they should have an or with undefined. Also, if the first 2 parameters are required, then optionsObject is required and shouldn't be optional. Also I think there should be some documentation about what the parameters mean

Reply:
Per conversations with KJ, it may make sense when using a custom measurement_url to not send either the ID, secret, or both.  In this case, they may be added server-side before being forwarded to GA (if they want to forward them at all).

So we decided to make them optional, with a warning that events will be dropped if they are omitted on the default collection path.  I assume the warning will be added in a followup PR.

Reply:
So this is a tricky one. In the design doc, [we discussed](https://docs.google.com/document/d/1peiKZs0e7C7n5GS9C1MHzODkHLPGBb55uL8lcqCtXCE/edit#heading=h.2elzgjrj4jsf) why a user would override the measurement URL and if the API secret and measurement ID would be included as query parameters. If, for example, they had a custom endpoint that would take in the events and route them to different Google Analytic properties (each with their own measurement ID) then we definitely don't want to force them to pick a single measurement ID.

The idea we came up with is to keep the API secret and measurement ID as completely optional parameters. If included, they will set the appropriate query parameter during the network request.

Of course for "target" scoping we require a single measurement ID attached to the GA event processor so it can look for the correct object in the global model and extract all its parameters. This will be a documented trade-off; If they decide to forgo a measurement ID then we will trust them to use automatic parameters for persisting important event parameters beyond the defaults.